### PR TITLE
fix(asm): don't spill a stacktrace on logs on remote config connection errors

### LINKED ISSUE #4961: Disable datadog tracer library logs
### Summary of problem

Apologies if this is not the place to ask for help. Our application log is flooded with error messages from datadog. Please see below. Setting aside the error messges themselves is there a way to completely disable all the logs from datadog library?

```
Traceback (most recent call last):
  File "/usr/src/app/.venv/lib/python3.9/site-packages/ddtrace/internal/remoteconfig/client.py", line 407, in request
    response = self._send_request(payload)
  File "/usr/src/app/.venv/lib/python3.9/site-packages/ddtrace/internal/remoteconfig/client.py", line 248, in _send_request
    self._conn.request("POST", "v0.7/config", payload, self._headers)
  File "/usr/local/lib/python3.9/http/client.py", line 1285, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.9/http/client.py", line 1331, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.9/http/client.py", line 1280, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.9/http/client.py", line 1040, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.9/http/client.py", line 980, in send
    self.connect()
  File "/usr/local/lib/python3.9/http/client.py", line 946, in connect
    self.sock = self._create_connection(
  File "/usr/local/lib/python3.9/socket.py", line 844, in create_connection
    raise err
  File "/usr/local/lib/python3.9/socket.py", line 832, in create_connection
    sock.connect(sa)
OSError: [Errno 99] Cannot assign requested address
```

### Which version of dd-trace-py are you using?
1.7.0

### Which libraries and their versions are you using?

<details>
  <summary>`pip freeze`</summary>
  
</details>

### PR DESCRIPTION
**Pull request by juanjux**
## Description

With this change, OSErrors on the remote config `_send_request` function will be captured and only the exception text (and not the full traceback) will be logged.

Also, DDLogger will be used on that module instead of the generic Python one so the user can set the desired level.

## Relevant issue(s)

Fixes #4961

Signed-off-by: Juanjo Alvarez <juanjo.alvarezmartinez@datadoghq.com>

### CODE REVIEW

**Review by brettlangdon — DISMISSED** [review]

File: releasenotes/notes/fix-dont-log-remoteconfig-stacktrace-ab7d1db057233454.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fixes:
+  - |
+    Fixed full stacktrace being sent to the log on remote config connection errors.
```
Comment (P403n1x87) [inline releasenotes/notes/fix-dont-log-remoteconfig-stacktrace-ab7d1db057233454.yaml:3]:
```suggestion
    Fix full stacktrace being sent to the log on remote config connection errors.
```
and is there an appropriate scope we can prefix? cc @majorgreys @mabdinur

**Review by P403n1x87 — COMMENTED** [review]
nit on relnote, otherwise LGTM

File: releasenotes/notes/fix-dont-log-remoteconfig-stacktrace-ab7d1db057233454.yaml (hunk 1 above)
Comment (juanjux) [reply releasenotes/notes/fix-dont-log-remoteconfig-stacktrace-ab7d1db057233454.yaml]: Thanks, commited.

File: ddtrace/internal/remoteconfig/client.py [hunk 2]
```diff
@@ -264,11 +264,15 @@ def register_product(self, product_name, func=None):
...
-            self._conn.close()
+            try:
+                self._conn.request("POST", REMOTE_CONFIG_AGENT_ENDPOINT, payload, self._headers)
+                resp = self._conn.getresponse()
+                data = resp.read()
+            finally:
+                self._conn.close()
+        except OSError as e:
+            log.warning("Unexpected connection error in remote config client request: %s", str(e))  # noqa: G200
+            return None
```
Comment (avara1986) [inline ddtrace/internal/remoteconfig/client.py:275]: is two `try`s needed?
```python
    data = None
    try:
        self._conn.request("POST", REMOTE_CONFIG_AGENT_ENDPOINT, payload, self._headers)
        resp = self._conn.getresponse()
        data = resp.read()
    except OSError as e:
        log.warning("Unexpected connection error in remote config client request: %s", str(e))  # noqa: G200
    finally:
        self._conn.close()
    if not data:
        return None
    if resp.status == 404:
       .....
```

**Review by avara1986 — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

**Pull request merged**